shm/Ship.cpp

```diff
@@ -0,0 +1,95 @@
+#include "Ship.hpp"
+
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+
+#include "Alcohol.hpp"
+#include "Fruit.hpp"
+#include "Item.hpp"
+
+Ship::Ship()
+    : id_(0) {
+}
+
+Ship::Ship(int capacity, int maxCrew, int speed, const std::string& name, size_t id, std::shared_ptr<Time>& time)
+    : capacity_(capacity), maxCrew_(maxCrew), crew_(0), speed_(speed), name_(name), id_(id), time_(time) {
+    time_->addObserver(this);
+}
+
+Ship::Ship(int maxCrew, int speed, size_t id, std::shared_ptr<Time>& time)
+    : Ship(0, maxCrew, speed, "", id, time) {
+}
+
+Ship& Ship::operator-=(const size_t crew) {
+    if (crew > crew_) {
+        crew_ = 0;
+        return *this;
+    }
+    crew_ -= crew;
+    return *this;
+}
+
+Ship& Ship::operator+=(const size_t crew) {
+    if (crew_ + crew > maxCrew_) {
```

Comment:
I think it's better to compare crew_ and crew, because if we add crew_ and crew we risk going out of size_t range,which may result in gargabe value that can be lower than crew limit.